### DISCUSSION TITLE
Version 2.0 Will Be Released Under AGPLv3

### DISCUSSION THREAD

**Discussion by faisalman** [faisalman/ua-parser-js#674] [Announcements]
```
Hi all, 

As you can see from the latest changes in the repository, from now on, 
version 2.0 upwards will be licensed under AGPLv3. However, those who are 
still using version 1.0.x / 0.7.x won't be affected by this change and you 
can keep using them as usual as they are MIT licensed, and always will be.

For those who haven't heard, AGPLv3 is an OSI-approved open-source 
license from GNU that requires your codebase to be open to the user of 
your code (pretty much like GPL). In order to be able to use it under a 
closed-source codebase, I will also release a PRO license which gives 
you permission to use it as you like, with a reasonably small price.

I know this might be an unpopular decision, but as the author of UAParser.js 
I really want to see this project moves further and take a risk in a hope 
that this project will have a long-term future by having a sustainable 
financially while still being free & open-source.

Regards,
Faisal Salman
```

**Comment by slhck** [comment]
Thanks for announcing this change. It's understandable that you would like to find some revenue source for maintaining this open-source project. However, it would be great for users who want to upgrade/migrate to provide details on "pro" or commercial licensing terms. You have almost 12 million weekly downloads, so please consider publishing a bit more information about this. Right now it's a bit of an unknown what this change means. Thanks!

**Reply by faisalman** [reply]
Thanks! I'll certainly provide more additional info later

**Comment by dephiros** [comment]
@faisalman , thank you for your work
I am preparing our company for this change and have a question:
- Is there a timeline for the deprecation of v1(bug, security fixes...)?

**Reply by faisalman** [reply]
Currently we have no plan to deprecate v1. It might still get occassional updates, particularly for security/bugfix-related updates.

**Comment by JamesBream** [comment]
Hey @faisalman, is there a roadmap for the v2.0 release? Happy to support the project but didn't want to move over to v2 while it was still in beta.

**Comment by yandi-banyu** [comment]
Hello, where can I review the license for the paid version? Specifically, what defines as "commercial" and what multiple usage of personal license is allowable.

Thank you in advance

**Reply by faisalman** [reply]
Hi, currently the license hasn't specifically defines the term "commercial". For "personal" license, multiple usage is allowed as long as you don't use it for commercial purpose. You can review the license here: https://www.npmjs.com/package/@ua-parser-js/pro-personal?activeTab=code (see license.md file)

**Reply by yandi-banyu** [reply]
I see. Well, I suggest you should probably clarify a lot of the clause there. Anyway, for me it is too vague as what constitute as commercial usage and the AGPL license is too vague in the context of JS for me too. I wish you luck on this project as a fellow programmer!

**Reply by slhck** [reply]
Yeah, I believe these license terms are too vague:

> … may be used directly for commercial purpose …

The word "directly" could be interpreted as being commercial only when offering, e.g., a SaaS or software that makes money off of parsing user agents, "directly" using this library to fulfill its main purpose.

If, say, I was using this library only as part of another software offering, where user agent parsing is a small feature, then it could be debated whether this is "direct" use for commercial purposes, or merely indirect use (e.g., for security-related purposes that have no added value to the offering itself).

For instance, Redis recently began dual-icensing their code, with the [main commercial license](https://redis.io/legal/rsalv2-agreement/) prohibits making available the software in a manner where you "offer a product or service, the value of which entirely or primarily derives from the value of the Software".

In general, it appears like your license terms were not fully created or reviewed by a legal professional. I wholeheartedly recommend you seek legal advice and build upon existing projects that incorporate dual-licensing to make this easier for your end-users, and, hopefully attract pro customers.

Right now I am still a bit unclear about the meaning, therefore I will stay with the legacy code for our commercial solution (we use the library to provide coarse information about OS and browser version as metadata).
